client/network/src/protocol.rs

```diff
@@ -365,51 +335,21 @@ where
 			sc_peerset::Peerset::from_config(sc_peerset::PeersetConfig { sets })
 		};
 
-		let block_announces_protocol = {
-			let genesis_hash =
-				chain.hash(0u32.into()).ok().flatten().expect("Genesis block exists; qed");
-			let genesis_hash = genesis_hash.as_ref();
-			if let Some(fork_id) = fork_id {
-				format!(
-					"/{}/{}/block-announces/1",
-					array_bytes::bytes2hex("", genesis_hash),
-					fork_id
-				)
-			} else {
-				format!("/{}/block-announces/1", array_bytes::bytes2hex("", genesis_hash))
-			}
-		};
-
-		let legacy_ba_protocol_name = format!("/{}/block-announces/1", protocol_id.as_ref());
-
 		let behaviour = {
-			let best_number = info.best_number;
-			let best_hash = info.best_hash;
-			let genesis_hash = info.genesis_hash;
-
-			let block_announces_handshake =
-				BlockAnnouncesHandshake::<B>::build(roles, best_number, best_hash, genesis_hash)
-					.encode();
-
-			let sync_protocol_config = notifications::ProtocolConfig {
-				name: block_announces_protocol.into(),
-				fallback_names: iter::once(legacy_ba_protocol_name.into()).collect(),
-				handshake: block_announces_handshake,
-				max_notification_size: MAX_BLOCK_ANNOUNCE_SIZE,
-			};
-
 			Notifications::new(
 				peerset,
-				iter::once(sync_protocol_config).chain(
-					network_config.extra_sets.iter().zip(notifications_protocols_handshakes).map(
-						|(s, hs)| notifications::ProtocolConfig {
-							name: s.notifications_protocol.clone(),
-							fallback_names: s.fallback_names.clone(),
-							handshake: hs,
-							max_notification_size: s.max_notification_size,
-						},
-					),
-				),
+				iter::once(notifications::ProtocolConfig {
+					name: block_announces_protocol.notifications_protocol.clone(),
+					fallback_names: block_announces_protocol.fallback_names.clone(),
+					handshake: block_announces_protocol.handshake.as_ref().unwrap().to_vec(),
+					max_notification_size: block_announces_protocol.max_notification_size,
+				})
```

Comment:
I assume we still have the requirement that this is the first protocol? If yes, could we leave some comment for now?